### ISSUE TITLE
Rename ovewrite

### ISSUE THREAD

**Issue by dtheodor** [microsoft/AzureStorageExplorer#3811]
As stated in the [ADLS Gen2 REST API](https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create), a create/rename operation can overwrite the existing file.

> Create or rename a file or directory. By default, the destination is overwritten[...]

As far as I can tell, this is impossible to do with the storage explorer. Running version 1.16.0.

**Comment by MRayermannMSFT** [comment]
@dtheodor is there a reason you can't just first delete the destination and then perform the rename? We'd like to stay on the safe side of things and not have renames be able to do overwrites.

**Comment by dtheodor** [comment]
Yes, the reason is that I want the overwrite to happen atomically. I am overwriting a directory, and my application depends on the directory always being there, I cannot afford it to be missing. An application-based workaround is practically impossible.

On further investigation this deficiency is also present in the REST API: although it advertises destination overwrite, it fails to do that with a similar error on 'legacy' mode, and it always moves the directory inside the already existing one on 'posix' mode. Maybe this is an issue that needs to be raised with the "server" team, but I have no way to contact them. These issues https://github.com/microsoft/AzureStorageExplorer/issues/1071 https://github.com/microsoft/AzureStorageExplorer/issues/2308 are related to this behavior.

**Comment by MRayermannMSFT** [comment]
@dtheodor

We can raise the issue with the platform team, and you can try raising this issue with Azure support, but it does sound like based on what you're saying that this request isn't possible right now.

You might need to rearchitect your application until the platform team can improve the API. =/
